src/dmd/dclass.d

```diff
@@ -446,6 +446,7 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
             *poffset = 0;
         while (cd)
         {
+            assert(cd.baseClass || cd.semanticRun >= PASS.semanticdone || cd.isInterfaceDeclaration());
```

Comment:
looks like this doesn't pass (in doc mode?)

Reply:
This is here to catch other bugs that crop up when semantic on classes are not done eagerly enough.

Reply:
FYI, this was the condition used to call `dsymbolSemantic(cd)` in this function prior to making this pure.  The assert just restores the check to catch any further bad cases.